llvm/lib/Target/WebAssembly/WebAssemblyISelLowering.cpp

```diff
@@ -3383,15 +3383,59 @@ static SDValue TryMatchTrue(SDNode *N, EVT VecVT, SelectionDAG &DAG) {
   return DAG.getZExtOrTrunc(Ret, DL, N->getValueType(0));
 }
 
+static SDValue
+combineVectorSizedSetCCEquality(SDNode *N, TargetLowering::DAGCombinerInfo &DCI,
+                                const WebAssemblySubtarget *Subtarget) {
+
+  SDLoc DL(N);
+  SDValue X = N->getOperand(0);
+  SDValue Y = N->getOperand(1);
+  EVT VT = N->getValueType(0);
+  EVT OpVT = X.getValueType();
+
+  ISD::CondCode CC = cast<CondCodeSDNode>(N->getOperand(2))->get();
+  SelectionDAG &DAG = DCI.DAG;
+  // We're looking for an oversized integer equality comparison.
+  if (!OpVT.isScalarInteger() || !OpVT.isByteSized() || OpVT != MVT::i128 ||
+      !Subtarget->hasSIMD128())
+    return SDValue();
```

Comment:
Oh this only works for SETEQ right? I don't think we can do this transform for arbitrary condition codes, e.g.

`i128 x < i128 y` can't be transformed to `all_true (v16i8 x < v16i8 y)`

I think we need to check for `isIntEqualitySetCC` here, and it would be good to add a negative test case for another cond code to make sure we don't match it accidentally